### PR TITLE
Issue 4778 - [RFE] Schedule execution of "compactdb" at specific date…

### PR DESCRIPTION
**Pull request by bsimonova**
…/time

Description:
Added additional tests to check invalid and no compaction value for
nsslapd-db-compactdb-interval.

Relates: https://github.com/389ds/389-ds-base/issues/4778

Reviewed by: ???

### CODE REVIEW

File: dirsrvtests/tests/suites/config/compact_test.py [hunk 1]
```diff
@@ -72,6 +74,62 @@ def test_compaction_interval_and_time(topo):
...
+    inst = topo.ms["supplier1"]
+    msg = 'value 2147483650 for attr nsslapd-db-compactdb-interval is greater than the maximum 2147483647'
+    config = DatabaseConfig(inst)
+
+    try:
+        config.set([('nsslapd-db-compactdb-interval', '2147483650'), ('nsslapd-db-compactdb-time', '00:01')])
+    except ldap.UNWILLING_TO_PERFORM as e:
+        log.info('Got expected error: {}'.format(str(e)))
+        assert msg in str(e)
+        time.sleep(6)
```
Comment (mreynolds389) [inline dirsrvtests/tests/suites/config/compact_test.py]: Can you reduce this sleep time?  Maybe 3 seconds?  These "sleeps" add up when running the entire test suite.

Comment (bsimonova) [reply dirsrvtests/tests/suites/config/compact_test.py]: I reduced the sleep time, please review. Thanks!

**Comment by bsimonova** [comment]
@mreynolds389 
I updated the sleep time, please review. Thanks!

File: dirsrvtests/tests/suites/config/compact_test.py [hunk 2]
```diff
@@ -72,6 +76,64 @@ def test_compaction_interval_and_time(topo):
     inst.deleteErrorLogs(restart=False)
 
 
+@pytest.mark.ds4778
+@pytest.mark.bz1748441
+@pytest.mark.skipif(ds_is_older("1.4.3.23"), reason="Not implemented")
+def test_no_compaction(topo):
```
Comment (tbordaz) [inline dirsrvtests/tests/suites/config/compact_test.py:82]: Something not clear to me. compaction occurs on checkpointing (via checkpointing thread dynamic) and on compact task.
When sleeping 3 seconds are we waiting for the checkpointing to occur ? But if there is no update, is the checkpointing still trying to compact ?
I would rather expect to trigger the compact via the task, then check the message in the error log. But here I see no task.

Comment (bsimonova) [reply dirsrvtests/tests/suites/config/compact_test.py:82]: @tbordaz 
I took the test_compaction_interval_and_time from Mark which sets nsslapd-db-compactdb-interval and nsslapd-db-compactdb-time. Then it waits and checks logs that compaction occurred. 
What I did in my test was that I set nsslapd-db-compactdb-interval to 0, which should not trigger any compaction and then I check the logs to confirm there was none. I just decreased the time.sleep as Mark suggested.
So I would say yes, we are waiting for the checkpointing if I understand correctly.

Comment (tbordaz) [reply dirsrvtests/tests/suites/config/compact_test.py:82]: @bsimonova , sorry my fault. Checkpointing thread runs periodically (2.5sec) and if nsslapd-db-compactdb-interval is not null it calls a bdb_compact that logs "Compacting databases" and calls database compaction. So even if database compaction does nothing (no updates), the log is present.
Sorry for the noise

**Review by tbordaz — APPROVED** [review]

**Pull request merged**